tempesta_fw/ss_skb.c

```diff
@@ -806,12 +806,7 @@ __skb_fragment(struct sk_buff *skb_head, struct sk_buff *skb, char *pspt,
 		d_size = skb_frag_size(frag);
 		offset = pspt - (char *)skb_frag_address(frag);
 
-		/*
-		 * @pspt can be the end of a frag, but it can also be a start of
-		 * another fragment. Not necessary, that both frags are
-		 * neighbours.
-		 */
-		if (offset >= 0 && offset < d_size) {
+		if (offset >= 0 && offset <= d_size) {
```

Comment:
The original condition was introduced in 0f04bfd29372ac722f818af3874e1bba3f1ef865 , which also removed the special handling of `offset == d_size`. `offse == d_size` means that `pspt` is just *after* the current fragment, so why should we split the current fragment?

Reply:
Split works like adding a new fragment